source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,182 @@
+      const QGauss<dim-1> quadrature_formula (this->introspection().polynomial_degree.velocities+1);
+   
+      FEFaceValues<dim> fe_face_values (this->get_mapping(),
+                                        this->get_fe(),
+                                        quadrature_formula,
+                                        update_gradients |
+                                        update_quadrature_points |
+                                        update_JxW_values);
+         
+      std::vector<Tensor<1,dim>> velocities (fe_face_values.n_quadrature_points);
+
+      // Vector to store the integral of the second invariant of the strain rate
+      // over the surface.
+      double local_second_invariant_of_strain_rate_integral = 0.0;
+      double local_surface_area_integral = 0.0;
+
+      std::vector<Tensor<1,dim>> local_second_invariant_of_strain_rate(fe_face_values.n_quadrature_points);
+      std::vector<Tensor<1,dim>> local_surface_area(fe_face_values.n_quadrature_points); 
+      std::vector<std::pair<double,double>> local_second_invariant_of_strain_rate_and_corresponding_area; 
+      std::vector<SymmetricTensor<2,dim>> strain_rate (quadrature_formula.size());
+ 
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned() && cell->at_boundary())
+          {
+            unsigned int face_idx = numbers::invalid_unsigned_int;
+            bool at_upper_surface = false;
+            {
+              for (unsigned int f=0; f<GeometryInfo<3>::faces_per_cell; ++f)
+                {
+                  if (cell->at_boundary(f) && cell->face(f)->boundary_id() == top_boundary_id)
+                    {
+                      // If the cell is at the top boundary, assign face_idx.
+                      face_idx = f;
+                      at_upper_surface = true;
+                      break;
+                    }
+                }
+
+            }
+            if (at_upper_surface)
+              {
+                fe_face_values.reinit(cell, face_idx);
+                fe_face_values[this->introspection().extractors.velocities].get_function_symmetric_gradients (this->get_solution(), strain_rate);
+              
+                for (unsigned int q = 0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    //Merge so column 1 is strain rate invariants and column 2 are the corresponding areas
+                    local_second_invariant_of_strain_rate[q] = std::max(std::sqrt(std::fabs(second_invariant(deviator(strain_rate[q])))),min_strain_rate);
+                    local_surface_area[q] = fe_face_values.JxW(q);
+                    
+                    // second invariant of strain rate over the whole surface
+                    local_second_invariant_of_strain_rate_integral = local_second_invariant_of_strain_rate[q] * local_surface_area[q];
+                    local_surface_area_integral += local_surface_area[q];
```

Comment:
This is not inherently wrong, but it's inefficient. You are in essence packing information into the `local_second_invariant_of_strain_rate` and `local_surface_area` arrays by writing into the `[q]`th entries in the first two lines. In the second set of two lines, you then use the information so stored, but after that, you never access the entries of these arrays again and simply overwrite these values on the next cell that happens to be on the boundary.

A more efficient way would be to just not put things into an array to begin with, but instead do something like the following:
```suggestion
                    const double local_second_invariant_of_strain_rate = std::max(std::sqrt(std::fabs(second_invariant(deviator(strain_rate[q])))),min_strain_rate);
                    const double local_surface_area[q] = fe_face_values.JxW(q);
                    
                    // second invariant of strain rate over the whole surface
                    local_second_invariant_of_strain_rate_integral += local_second_invariant_of_strain_rate * local_surface_area;
                    local_surface_area_integral += local_surface_area;
```
Then you can get rid of the array declarations above.